backend/monitoring_integrations/router.py

```
@@ -0,0 +1,72 @@
+    them to the appropriate adapter.
+    """
+    body = await request.json()
+    adapter = _ADAPTERS.get(provider.lower())
+    if not adapter:
+        raise HTTPException(status_code=404, detail=f"Provider not supported: {provider}")
+
+    try:
+        alerts = adapter.parse_payload(body)
+    except AdapterParseError as e:
+        logger.warning(f"Adapter parse error for {provider}: {e}")
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)